modules/thermal_hydraulics/src/components/HeatTransferFromHeatStructure3D1Phase.C

```diff
@@ -237,7 +237,9 @@ HeatTransferFromHeatStructure3D1Phase::addVariables()
       false, _P_hf_name, getTHMProblem().getFlowFEType(), _flow_channel_subdomains);
 
   _P_hf_fn_name = getParam<FunctionName>("P_hf");
-  getTHMProblem().addFunctionIC(_P_hf_name, _P_hf_fn_name, _flow_channel_subdomains);
+
+  if (!_app.isRestarting())
+    getTHMProblem().addFunctionIC(_P_hf_name, _P_hf_fn_name, _flow_channel_subdomains);
```

Comment:
I dont see an auxkernel to respect this parameter if it has changed during the restart